src/_pytest/assertion/rewrite.py

```diff
@@ -317,7 +317,7 @@ def _write_pyc_fp(
     flags = b"\x00\x00\x00\x00"
     fp.write(flags)
     # as of now, bytecode header expects 32-bit numbers for size and mtime (#4903)
```

Comment:
Sorry I missed it before, but this comment seems to negate this change -- it takes the lower 32 bits while st_mtime_ns needs at least 64 bits, e.g.

```pycon
>>> import os
>>> st = os.stat('/home')
>>> print(st.st_mtime_ns)
1669828167393028173
>>> print(st.st_mtime_ns & 0xFFFFFFFF)
23537741
>>> print(int(st.st_mtime) & 0xFFFFFFFF)
1669828167
```

Basically, the pyc timestamp format is 32 bits, and we are trying to stay compatible with it (even though it's not absolutely necessary as we do our own loading). The same problem occurs with python itself, not just pytest.

See:
https://github.com/python/cpython/issues/121376
https://discuss.python.org/t/change-pyc-file-format-to-record-more-accurate-timestamp-and-other-information/57815

Also related: #11418

Reply:
You're correct, this is more complex than I expected, I used to run the same test without pytest, but it didn't reproduce somehow, so I thought it's only caused by pytest. Actually, the similar issue still exists in python's import system, and pytest's cache rewrite inherited the same issue. Looks like no proper solution for the issue until now, perhaps we should close the PR.